### DISCUSSION TITLE
Idea: Move from YAML config to PHP?

### DISCUSSION THREAD

**Discussion by ThomasLandauer** [Codeception/Codeception#6007] [Ideas]
Symfony is moving from YAML to PHP configuration: https://github.com/symfony/symfony/issues/37186

So I'm wondering if this would be a good idea for Codeception too?

I'm not sure :-) However, it would solve these issues:
* https://github.com/Codeception/Codeception/issues/5642
* https://github.com/Codeception/Codeception/issues/5661
* https://github.com/Codeception/Codeception/issues/5984

**Comment by DavertMik** [comment]
Let's wait for Kubernetes to move to PHP configuration and then we can probably migrate too :rofl: 

But yeah, Codeception & Symfony is quite different. And even I like the idea of configuring things in PHP via DSL I know that implementing PHP DSL for configuration can be a very expensive toy to develop. Unfortunately, I don't have time for it right now.

But if there are volunteers to work on this idea - I'm ok to guide the process

**Comment by ThomasLandauer** [comment]
Symfony continues to improve the support for PHP config from version to version. Meanwhile Codeception is the only package I'm using that requires `symfony/yaml`.

**Reply by DavertMik** [reply]
Don't you work with GitHub actions? They also use YAML format
And YAML is more poplar than PHP configs.
